### PR TITLE
Feat/xdsl warning incident

### PR DESCRIPTION
**Pull request by JeremyDec**
MBE-339

Need merge of new ovh-api-services route : 
- [x] https://github.com/ovh-ux/ovh-api-services/pull/153

### CODE REVIEW

**Comment by antleblanc** [comment]
@JeremyDec [ovh-api-services](https://github.com/ovh-ux/ovh-api-services/releases/tag/v6.10.0) has been released!

File: client/app/telecom/pack/xdsl/access/pack-xdsl-access.controller.js [hunk 1]
```diff
@@ -90,26 +93,26 @@ angular.module('managerApp').controller('XdslAccessCtrl', class XdslAccessCtrl {
...
+    this.$rootScope.$on('accessDiagnosticDetails:launch', () => this.launchDiagnostic());
 
     this.$rootScope.$on('accessDiagnosticDetails:get', () => {
+      let launchDiagnostic;
       if (!this.accessDiagnostic && !this.$scope.loaders.accessDiagnosticLaunched) {
-        this.launchDiagnostic();
+        launchDiagnostic = this.launchDiagnostic();
       }
       this.$rootScope.$broadcast('accessDiagnosticDetails:arrived', this.accessDiagnostic);
+      return launchDiagnostic;
```
Comment (FredericEspiau) [inline client/app/telecom/pack/xdsl/access/pack-xdsl-access.controller.js]: Hmm `$on` returns a function to unregister the event, so this wouldn't have any effect

File: client/app/telecom/pack/xdsl/access/pack-xdsl-access.controller.js [hunk 2]
```diff
@@ -338,4 +349,24 @@ angular.module('managerApp').controller('XdslAccessCtrl', class XdslAccessCtrl {
...
+  getIncident() {
+    return this.OvhApiXdsl.Incident().v6().get({
+      serviceName: this.$stateParams.serviceName,
+    }).$promise
+      .then(() => {
+        this.hasIncidentOccured = true;
+      })
+      .catch((error) => {
+        const errorMessage = _.get(error, 'data.message', '');
+        if (errorMessage === XDSL_NO_INCIDENT_MESSAGE) {
```
Comment (marie-j) [inline client/app/telecom/pack/xdsl/access/pack-xdsl-access.controller.js]: Could we rely  on checking the `error.status === 404` instead ? I don't know if that makes sense , let's suppose the API error message changes, we have to change it here also

File: client/app/telecom/pack/xdsl/access/pack-xdsl-access.controller.js [hunk 3]
```diff
@@ -90,26 +93,26 @@ angular.module('managerApp').controller('XdslAccessCtrl', class XdslAccessCtrl {
       Until the task is finished, the diagnostic we get from
       the API is IRRELEVANT and should not be displayed to the user.
     */
-    this.$rootScope.$on('accessDiagnosticDetails:launch', () => {
-      this.launchDiagnostic();
-    });
+    this.$rootScope.$on('accessDiagnosticDetails:launch', () => this.launchDiagnostic());
 
     this.$rootScope.$on('accessDiagnosticDetails:get', () => {
+      let launchDiagnostic;
```
Comment (marie-j) [inline client/app/telecom/pack/xdsl/access/pack-xdsl-access.controller.js]: Not yours but do you think we can listen to / emit from`$scope` instead ?

File: client/app/telecom/pack/xdsl/access/pack-xdsl-access.html [hunk 4]
```diff
@@ -29,15 +29,30 @@
...
+                                            <oui-message data-type="warning">
+                                                <p data-translate="xdsl_details_diagnostic_known_incident"></p>
+                                                <span class="text-nowrap"
+                                                      data-translate="xdsl_details_diagnostic_known_incident_details"></span>
+                                                <a class="oui-link"
+                                                   data-ng-href="{{:: XdslAccess.URLS.xdslIncident }}"
+                                                   rel="noopener"
+                                                   target="_blank">
+                                                    <span data-ng-bind="XdslAccess.URLS.xdslIncident"></span>
+                                                </a>
```
Comment (marie-j) [inline client/app/telecom/pack/xdsl/access/pack-xdsl-access.html]: No `oui-icon-external_link` ?

File: client/app/telecom/pack/xdsl/access/pack-xdsl-access.controller.js (hunk 2 above)
Comment (JeremyDec) [reply client/app/telecom/pack/xdsl/access/pack-xdsl-access.controller.js]: Yes I would prefer rely on the `404` code, but the API returns two different errors for the same error code. I asked them to change this but they won't as this is made this way

File: client/app/telecom/pack/xdsl/access/pack-xdsl-access.html (hunk 4 above)
Comment (JeremyDec) [reply client/app/telecom/pack/xdsl/access/pack-xdsl-access.html]: Sure :)

File: client/app/telecom/pack/xdsl/access/pack-xdsl-access.controller.js (hunk 3 above)
Comment (JeremyDec) [reply client/app/telecom/pack/xdsl/access/pack-xdsl-access.controller.js]: Good point, I'd thought about this too, but as expected, `$rootScope` catches the event where `$scope` doesn't.

The emitter is located here : https://github.com/ovh-ux/ovh-manager-telecom/blob/master/client/app/telecom/pack/xdsl/access/diagnostic/details/pack-xdl-access-diagnostic-details.controller.js#L18 => It is the `$rootScope` :).

The whole event broadcast logic should be improved so, let's tackle this in another task (Creating a JIRA)

**Comment by jleveugle** [comment]
Hello @JeremyDec,
Your pull request is now closed due to inactivity.
Feel free to re-open it if you have any update.
Anna Bot for OVH UX <3

**Comment by jleveugle** [comment]
Hello @JeremyDec,
Your pull request is now closed due to inactivity.
Feel free to re-open it if you have any update.
Anna Bot for OVH UX <3

**Pull request merged**